sn_networking/src/lib.rs

```diff
@@ -631,9 +636,33 @@ impl Network {
     /// Put `Record` to network
     /// optionally verify the record is stored after putting it to network
     pub async fn put_record(&self, record: Record, verify_store: bool) -> Result<()> {
+        if verify_store {
+            self.put_record_with_retries(record).await
+        } else {
+            self.put_record_once(record, false).await
+        }
+    }
+
+    /// Put `Record` to network
+    /// Verify the record is stored after putting it to network
+    /// Retry up to `PUT_RECORD_RETRIES` times if we can't verify the record is stored
+    async fn put_record_with_retries(&self, record: Record) -> Result<()> {
+        let mut retries = 0;
+        while retries < PUT_RECORD_RETRIES {
+            let res = self.put_record_once(record.clone(), true).await;
```

Comment:
the flag true here will trigger a multiple attempts of `get_record` internally?
which means the total completion time could be: ` PUT_RECORD_RETRIES  * GET_ATTEMPTS * interval` ?

Reply:
Worse case scenario yes. But only if the error matches `FailedToVerifyRecordWasStored` all other errors or success will be immediately returned.